### PR TITLE
`Development`: Fix wrong result subscription for exam exercises

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
We noticed some log messages saying:
```
User with login '' tried to access the protected topic: /topic/exercise/14238/newResults
User with login '' tried to access the protected topic: /topic/exercise/10200/newResults
```

This PR fixes the cause of these warnings.

### Description
When opening the exam summary page, a subscription to result updates is initiated. The client incorrectly used the non-personal topic instead of the personal topic. This is now fixed.

### Steps for Testing
You can best test this locally or on a test server with access to the logs.

As a student, view your exams results summary. Then check the server and make sure none of the warnings as above are present.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance even for very large courses with more than 2000 students.
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance even for very large courses with more than 2000 students.
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Performance Tests
- [ ] Test 1
- [ ] Test 2

## Summary by CodeRabbit

- **New Features**
	- Introduced an `instructorView` property to enhance the exam result summary display for instructors.
	- Added a `personalParticipation` property to better manage participation visibility based on the instructor view.

- **Bug Fixes**
	- Streamlined authorization checks for user subscriptions to exercise results, improving logic clarity and efficiency.

- **Documentation**
	- Updated component signatures to reflect new input properties for better integration and usage.

### CODE REVIEW

**Review by JohannesStoehr — APPROVED** [review]
Code

File: src/main/webapp/app/exam/participate/summary/exercises/programming-exam-summary/programming-exam-summary.component.ts [hunk 1]
```diff
@@ -39,6 +39,8 @@ export class ProgrammingExamSummaryComponent implements OnInit {
 
     @Input() isAfterResultsArePublished?: boolean = false;
 
+    @Input() instructorView?: boolean = false;
```
Comment (edkaya) [inline src/main/webapp/app/exam/participate/summary/exercises/programming-exam-summary/programming-exam-summary.component.ts:42]: I think we have decided to use new `input()` and `output()` decorators
```suggestion
    instructorView = input<boolean>(false);
```

**Review by edkaya — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/exam/participate/summary/exercises/programming-exam-summary/programming-exam-summary.component.ts (hunk 1 above)
Comment (Strohgelaender) [reply src/main/webapp/app/exam/participate/summary/exercises/programming-exam-summary/programming-exam-summary.component.ts:42]: That's true, but for this I'd need to change the whole input chain which is quite big and way out for scope for this PR.

**Review by edkaya — APPROVED** [review]
changes lgtm

**Review by krusche — APPROVED** [review]

**Pull request merged**